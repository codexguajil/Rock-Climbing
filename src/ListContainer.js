import React from 'react';
import Description from './Description';

export default function ListContainer (props) {

  // let routeNames = props.areaAndRoutes[props.areaName].classicRoutes.map((route) => {
  //   return route.name
  // })

  let matchedAreas = Object.keys(props.areasAndRoutes).reduce((acc, area) => {
    if (props.areaName.includes(area)) {
     acc.push(props.areasAndRoutes[area])
    }
    return acc;
  }, [])

  

  let matchedType = props.areaType.reduce((acc, selectedType) => {
    matchedAreas.forEach((uniqueArea) => {
      uniqueArea.classicRoutes.forEach((route) => {
        if (route.type.includes(selectedType)) {
          acc.push(route)
        }
      })
    })

    return acc;
  }, []);



  return (
      <ul>
        {
          
        matchedType.map((type) => {
          return (
            <Description matchedType={type} 
            />
          )
        })
        
      }
      </ul>
    )
}

