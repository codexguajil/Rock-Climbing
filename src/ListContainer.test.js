import React from 'react';
import ListContainer from './ListContainer';
import { shallow } from 'enzyme';

const mockFilterAreas = ["Canon City", "Boulder Canyon"];

const mockFilterTypes = ["toprope"];

const mockColoradoClimbingAreas = {
    "Canon City": {
        "milesFromDenver": 119,
        "id": 34,
        "typesOfClimbing": ["trad", "sport", "toprope", "boulder"],
        "classicRoutes": [
            {
                "name": "Enterprise",
                "id": 35,
                "grade": "5.09+",
                "type": ["sport"],
                "link": "https://www.mountainproject.com/route/105824687/enterprise"
            },
            {
                "name": "Trick-or-Treat",
                "id": 42,
                "grade": "5.11a/b",
                "type": ["toprope"],
                "link": "https://www.mountainproject.com/route/105759012/trick-or-treat"
            }
        ]
    },
    "Boulder Canyon": {
        "milesFromDenver": 31.5,
            "id": 12,
            "typesOfClimbing": ["trad", "sport", "toprope", "boulder", "aid"],
            "classicRoutes": [
                {
                    "name": "North Face Center",
                    "id": 13,
                    "grade": "5.07+",
                    "type": ["trad"],
                    "link": "https://www.mountainproject.com/route/105748093/north-face-center"
                },
                {
                    "name": "Empor",
                    "id": 14,
                    "grade": "5.07+",
                    "type": ["trad"],
                    "link": "https://www.mountainproject.com/route/105748142/empor"
                }
            ]
        }
    }

const mockFilterTypesSearch = [];
const mockFilterAreasSearch = ["Boulder Canyon"];

describe('ListContainer', () => {

    
    it('should produce a collection of climbing routes if user selects type & area', () => {
    const wrapper = shallow(
        <ListContainer filterAreas={mockFilterAreas}
                       areasAndRoutes={mockColoradoClimbingAreas}
                       filterTypes={mockFilterTypes}/>
    );
        expect(wrapper).toMatchSnapshot(); 
    });

   it('should produce a list of climbing routes that are within the area that the user types in the search', () => {
    const wrapper = shallow(
        <ListContainer filterAreas={mockFilterAreasSearch}
                       areasAndRoutes={mockColoradoClimbingAreas}
                       filterTypes={mockFilterTypesSearch}/>
    );
        expect(wrapper).toMatchSnapshot(); 
    })

});
