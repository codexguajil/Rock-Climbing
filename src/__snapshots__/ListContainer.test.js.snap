// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer should produce a collection of climbing routes if user selects type & area 1`] = `
<ul
  className="climbing-route-list-container"
>
  <Description
    matchedType={
      Object {
        "grade": "5.11a/b",
        "id": 42,
        "link": "https://www.mountainproject.com/route/105759012/trick-or-treat",
        "name": "Trick-or-Treat",
        "type": Array [
          "toprope",
        ],
      }
    }
  />
</ul>
`;

exports[`ListContainer should produce a list of climbing routes that are within the area that the user types in the search 1`] = `
<ul
  className="climbing-route-list-container"
/>
`;
